haystack/nodes/retriever/dense.py

```diff
@@ -453,15 +448,6 @@ def retrieve_batch(
         if batch_size is None:
             batch_size = self.batch_size
 
-        if isinstance(filters, list):
-            if len(filters) != len(queries):
```

Comment:
The lists need to be of the same length so I would prefer to keep this check. Could you please explain why we should remove it? Is it checked somewhere else already?

Reply:
These checks were moved down to `BaseDocumentStore.query_by_embedding_batch` and `SearchEngineDocumentStore.query_by_embedding_batch` I hope I've checked any occurence. But I'll double check and add a comment on each removal to point out where the check is now happening.